File: all-submissions/0408-valid-word-abbreviation/solution.java
class Solution {
    public boolean validWordAbbreviation(String word, String abbr) {
        char[] wordArr = word.toCharArray();
        char[] abbrArr = abbr.toCharArray();

        int i = 0;
        int j =0;
        while(i<wordArr.length && j<abbrArr.length) {
            if(wordArr[i] == abbrArr[j]) {
                i++;
                j++;
                continue;
            }
            if(!Character.isDigit(abbrArr[j])) return false;
            StringBuilder number = new StringBuilder();
            while(j<abbrArr.length && Character.isDigit(abbrArr[j])){
                number.append(abbrArr[j]);
                j++;
            }
            if(number.charAt(0) == '0') return false;
            int num = Integer.parseInt(number.toString());
            i=i+num;
        }
        if(i==wordArr.length && j==abbrArr.length) return true;
        return false;
    }
}

File: all-submissions/0408-valid-word-abbreviation/solution.py
class Solution:
    def validWordAbbreviation(self, word: str, abbr: str) -> bool:
        i = 0
        j = 0
        while i<len(word) and j<len(abbr):
            if word[i].isalpha() and abbr[j].isalpha():
                if word[i]==abbr[j]:
                    i+=1
                    j+=1
                    continue
                else:
                    return False
            elif abbr[j] == "0":
                return False
            else:
                num = 0
                while j<len(abbr) and abbr[j].isdigit():
                    num = num*10 + int(abbr[j])
                    j+=1
                i+=num
        return True if i== len(word) and j == len(abbr) else False

